fl4health/strategies/feddg_ga_with_adaptive_constraint.py

```diff
@@ -0,0 +1,268 @@
+from logging import INFO, WARNING
+from typing import Callable, Dict, List, Optional, Tuple, Union
+
+import numpy as np
+from flwr.common import MetricsAggregationFn, NDArrays, Parameters, ndarrays_to_parameters, parameters_to_ndarrays
+from flwr.common.logger import log
+from flwr.common.typing import FitRes, Scalar
+from flwr.server.client_proxy import ClientProxy
+
+from fl4health.parameter_exchange.parameter_packer import ParameterPackerAdaptiveConstraint
+from fl4health.strategies.aggregate_utils import aggregate_losses
+from fl4health.strategies.feddg_ga import FairnessMetric, FairnessMetricType, FedDgGa
+
+
+class FedDgGaAdaptiveConstraint(FedDgGa):
+    def __init__(
+        self,
+        *,
+        fraction_fit: float = 1.0,
+        fraction_evaluate: float = 1.0,
+        min_fit_clients: int = 2,
+        min_evaluate_clients: int = 2,
+        min_available_clients: int = 2,
+        evaluate_fn: Optional[
+            Callable[
+                [int, NDArrays, Dict[str, Scalar]],
+                Optional[Tuple[float, Dict[str, Scalar]]],
+            ]
+        ] = None,
+        on_fit_config_fn: Optional[Callable[[int], Dict[str, Scalar]]] = None,
+        on_evaluate_config_fn: Optional[Callable[[int], Dict[str, Scalar]]] = None,
+        accept_failures: bool = True,
+        initial_parameters: Parameters,
+        fit_metrics_aggregation_fn: Optional[MetricsAggregationFn] = None,
+        evaluate_metrics_aggregation_fn: Optional[MetricsAggregationFn] = None,
+        initial_loss_weight: float = 1.0,
```

Comment:
I think this is missing from the docstring in the constructor

Reply:
Yup, good call.